### PR TITLE
Enhance Views for User Profile and Posts

### PR DESCRIPTION
**Pull request by stino-x**
Description:
Hi team,

Iheagwara Augustine here. I've made some improvements to our views, specifically focusing on the user profile and posts. Here's a breakdown of the changes:

Profile Page (app/views/users/show.html.erb):

Properly indented the HTML for better readability.
Updated the layout and styling for a more user-friendly experience.
Included user details and a list of posts.
Posts Index Page (app/views/posts/index.html.erb):

Ensured correct display of user details along with each post.
Improved the layout of posts, comments, and buttons.
Other General Enhancements:

Checked and adjusted the styling and structure in various views.
These changes aim to enhance the overall user experience and maintain a consistent and appealing design across our application.

Testing:
I've tested these changes locally, ensuring that user profiles and posts are displayed as expected. No issues were identified during testing.

Any feedback or suggestions are welcome! Let me know if further adjustments are needed.

Best,
Iheagwara Augustine

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,59 @@
...
+    </div>
+
+    <div class="bio">
+      <h3 class="bio-title">Bio</h3>
+      <p class="bio-content">
+        <%= @user.bio %>
+      </p>
+    </div>
+
+    <% @user.posts.each do |post| %>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/show.html.erb]: - [ ] Here you are supposed to show the user's recent posts i.e 3 posts, currently, it shows all post kindly fix this

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1 +1,62 @@
...
+          <div class="commentsandlikes">
+            <h5 class="sign">Comments:</h5>
+            <span class="numberofcomments"><%= post.comments_counter %></span>
+            <h5>Likes:</h5>
+            <span class="numberoflikes"><%= post.likes_counter %></span>
+          </div>
+        </div>
+
+        <ul class="comment-list">
+          <% post.comments.each do |comment| %>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/index.html.erb]: - [ ] According to project guidelines, here you are supposed to show recent comments which in your model you limit to 5, currently all post comments show on post index page, kindly fix this

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @stino-x  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (stino-x) [reply app/views/posts/index.html.erb]: where in the requirements did you see that? i didnt see that

**Comment by bellom** [comment]
Hello @stino-x  👋,

_This is an additional review as a result of the second opinion request._

I'm afraid the CR is correct, however, more clearer explanation could have been given.

Here it is, let's take a look at the project requirement, screenshot below;
image
image

You will see in the wireframe that, you're supposed to show 3 recent posts and  5 recent comments, that's why you have created the methods in your model in the previous milestone. 

Also, you have been told in the project requirement to use methods created in the previous milestone;
image

I hope this is clear now.

kindly go ahead and implement the changes and ask for a review on your dashboard afterward.

Keep rocking 🚀!

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,62 @@
...
+            <h5 class="sign">Comments:</h5>
+            <span class="numberofcomments"><%= post.comments_counter %></span>
+            <h5>Likes:</h5>
+            <span class="numberoflikes"><%= post.likes_counter %></span>
+          </div>
+        </div>
+
+        <ul class="comment-list">
+          <% post.comments.order(created_at: :desc).limit(5).each do |comment| %>
+          <li class="comment">
```
Comment (ShadyShawkat) [inline app/views/posts/index.html.erb:36]: - [ ] [OPTIONAL] While you implemented this which has the right outcome. The desired method to use here is the method used in the model to get the last 5 comments which may look like this:

```ruby
<% post.recent_comments do |comment| %>
```

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1 +1,59 @@
...
+
+    <div class="bio">
+      <h3 class="bio-title">Bio</h3>
+      <p class="bio-content">
+        <%= @user.bio %>
+      </p>
+    </div>
+
+    <% @user.posts.order(created_at: :desc).limit(3).each do |post| %>
+      <div class="post">
```
Comment (ShadyShawkat) [inline app/views/users/show.html.erb:28]: - [ ] [OPTIONAL] While you implemented this which has the right outcome. The desired method to use here is the method used in the model to get the last 3 posts which may look like this:

```ruby
<% @user.recent_posts do |post| %>
```

**Review by ShadyShawkat — APPROVED** [review]
## Hi @stino-x, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bellom** [comment]
@ShadyShawkat - You are supposed to have asked for changes here instead of making it optional, @stino-x didn't use the methods created.

@stino-x - It would be nice to implement the optional changes from @ShadyShawkat.

**Pull request merged**